modules/elastic/src/main/scala/ch/epfl/bluebrain/nexus/kg/query/BaseElasticQueries.scala

```diff
@@ -0,0 +1,98 @@
+package ch.epfl.bluebrain.nexus.kg.query
+
+import cats.instances.string._
+import ch.epfl.bluebrain.nexus.commons.es.client.ElasticClient
+import ch.epfl.bluebrain.nexus.commons.http.HttpClient
+import ch.epfl.bluebrain.nexus.commons.types.search.{Pagination, QueryResults, Sort, SortList}
+import ch.epfl.bluebrain.nexus.kg.core.{ConfiguredQualifier, Qualifier}
+import ch.epfl.bluebrain.nexus.kg.core.domains.DomainId
+import ch.epfl.bluebrain.nexus.kg.core.IndexingVocab.PrefixMapping.rdfTypeKey
+import ch.epfl.bluebrain.nexus.kg.core.Qualifier._
+import ch.epfl.bluebrain.nexus.kg.core.organizations.OrgId
+import ch.epfl.bluebrain.nexus.kg.core.schemas.{SchemaId, SchemaName}
+import ch.epfl.bluebrain.nexus.kg.indexing.ElasticIndexingSettings
+import io.circe.Json
+
+abstract class BaseElasticQueries[F[_], Id](elasticClient: ElasticClient[F], settings: ElasticIndexingSettings)(
+    implicit
+    rs: HttpClient[F, QueryResults[Id]]) {
+
+  val ElasticIndexingSettings(_, _, base, baseVoc) = settings
+  protected val defaultSort                        = SortList(List(Sort(Sort.OrderType.Asc, "@id")))
+
+  implicit val orgIdQualifier: ConfiguredQualifier[OrgId]           = Qualifier.configured[OrgId](base)
+  implicit val domainIdQualifier: ConfiguredQualifier[DomainId]     = Qualifier.configured[DomainId](base)
+  implicit val schemaIdQualifier: ConfiguredQualifier[SchemaId]     = Qualifier.configured[SchemaId](base)
+  implicit val schemaNameQualifier: ConfiguredQualifier[SchemaName] = Qualifier.configured[SchemaName](base)
+  implicit val stringQualifier: ConfiguredQualifier[String]         = Qualifier.configured[String](baseVoc)
+  val deprecatedKey: String                                         = "deprecated".qualifyAsString
+  val publishedKey: String                                          = "published".qualifyAsString
+
+  protected val rdfType: String
+
+  protected def termsFrom(deprecated: Option[Boolean], published: Option[Boolean]): Seq[Json] = {
+    val deprecatedTerm = deprecated.map(d => term(deprecatedKey, d))
+    val publishedTerm  = published.map(p => term(publishedKey, p))
+
+    Seq(deprecatedTerm, publishedTerm).flatten
+  }
+
+  protected def query(terms: Json*): Json = {
+    matchAllQuery(Json.arr(terms :+ term(rdfTypeKey, rdfType): _*))
+  }
+
+  protected def term(key: String, value: Boolean): Json =
+    Json.obj(
+      "term" -> Json.obj(
+        key -> Json.fromBoolean(value)
+      )
+    )
+  protected def term(key: String, value: String): Json =
+    Json.obj(
+      "term" -> Json.obj(
+        key -> Json.fromString(value)
+      )
+    )
+
+  private def matchAllQuery(filterTerms: Json) =
+    Json.obj(
```

Comment:
I think you don't need all of that. You can have simply `Json.obj("query" -> Json.obj("bool" -> Json.obj("filter" -> Json.obj("bool" -> Json.obj("must" -> Json.arr(terms: _*))))))`